webapp/channels/src/components/advanced_create_post/advanced_create_post.tsx

```diff
@@ -996,9 +996,9 @@ class AdvancedCreatePost extends React.PureComponent<Props, State> {
     };
 
     handleUploadError = (err: string | ServerError, clientId?: string, channelId?: string) => {
-        let serverError = null;
-        if (typeof err === 'string' && err.length > 0) {
-            serverError = new Error(err);
+        let serverError = err;
+        if (typeof serverError === 'string') {
+            serverError = new Error(serverError);
```

Comment:
Making this consistent with the handling when creating a comment - https://github.com/mattermost/mattermost-server/blob/master/webapp/channels/src/components/advanced_create_comment/advanced_create_comment.tsx#L1077-L1080

Reply:
This was also a bug because if `err` wasn't a string, `serverError` would always remain `null`